nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/DataTypeUtils.java

```diff
@@ -469,21 +473,51 @@ public static RecordSchema inferSchema(final Map<String, Object> values, final S
             final RecordField recordField = new RecordField(key, inferredDataType, true);
             inferredFieldTypes.add(recordField);
 
-            final Object coercedValue = convertType(rawValue, inferredDataType, fieldName, charset);
-            coercedValues.put(key, coercedValue);
+            convertType(rawValue, inferredDataType, fieldName, charset);
         }
 
-        final RecordSchema inferredSchema = new SimpleRecordSchema(inferredFieldTypes);
-        return inferredSchema;
+        return new SimpleRecordSchema(inferredFieldTypes);
     }
 
     public static Record toRecord(final Object value, final String fieldName, final Charset charset) {
+        return toRecord(value, fieldName, charset, false);
+    }
+
+    private static Object covertObjectToRecord(final Object rawValue, final String key, final Charset charset) {
```

Comment:
typo - name is `covert...` instead of `convert...`
But we should not have a method named `convertObjectToRecord` that returns an Object that may be an array, may be a Record, or may be any other object that was provided. Need to either always return a Record or create a method name that reflects what is truly done.